### PR TITLE
Add DSN parsing

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Part of #1.

### CODE REVIEW

File: dsn.go [hunk 1]
```diff
@@ -13,21 +13,64 @@
...
-	AccessKey string
-	Endpoint  string
+	// POP config
+	POPAccessID  string
+	POPAccessKey string
+	POPURL       string
+	// Alisa config
+	AlisaAccessID  string
+	AlisaAccessKey string
+	AlisaProject   string
```
Comment (weiguoz) [inline dsn.go]: The `ENV` already carries the `Alisa*` properties.

Comment (tonyyang-svail) [reply dsn.go]: I see. So we don't need the Alisa* field?

Comment (weiguoz) [reply dsn.go]: No, we don't need these fields.

File: dsn.go [hunk 2]
```diff
@@ -13,21 +13,55 @@
...
 
-// Config is the deserialization of connect string
+import (
+	"fmt"
+	"net/url"
+	"regexp"
+)
+
+var (
+	reDSN = regexp.MustCompile(`^([a-zA-Z0-9_-]+):([=a-zA-Z0-9_-]+)@([:a-zA-Z0-9/_.-]+)\?([^/]+)$`)
```
Comment (shendiaomo) [inline dsn.go:23]: `[a-zA-Z0-9_-] ` is an equivalent of `\w`, should we use `\w` for better readability?

**Review by weiguoz — APPROVED** [review]
LGTM.
Please also take a look at @shendiaomo 's [suggestion](https://github.com/sql-machine-learning/goalisa/pull/9/files#r361574849).

**Pull request merged**